pkg/controller/core.oam.dev/v1alpha2/applicationrollout/applicationrollout_controller.go

```diff
@@ -123,19 +128,26 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (res reconcile.Result, retErr e
 			// continue to handle the previous resources until we are okay to move forward
 			targetAppRevisionName = appRollout.Status.LastUpgradedTargetAppRevision
 			sourceAppRevisionName = appRollout.Status.LastSourceAppRevision
+		} else {
+			// so that we don't think we are modified again
+			appRollout.Status.LastUpgradedTargetAppRevision = targetAppRevisionName
+			appRollout.Status.LastSourceAppRevision = sourceAppRevisionName
 		}
 		appRollout.Status.StateTransition(v1alpha1.RollingModifiedEvent)
 	}
 
-	// Get the source application
+	//TODO: handle deleting/abandoning state differently
```

Comment:
what does it mean? In which case we need this?

Reply:
the finalizer will stuck if the underlying resources (appRevision, etc) are deleted